src/Razor/src/Microsoft.VisualStudio.RazorExtension/SyntaxVisualizer/SyntaxVisualizerControl.xaml.cs

```diff
@@ -181,13 +251,82 @@ public void ShowGeneratedHtml()
             return;
         }
 
+        EnsureInitialized();
+
+        if (_languageServerFeatureOptions.UseRazorCohostServer)
+        {
+            if (_fileUriProvider.TryGet(_activeWpfTextView.TextBuffer, out var hostDocumentUri))
+            {
+                var request = new DocumentContentsRequest
+                {
+                    TextDocument = new TextDocumentIdentifier { DocumentUri = hostDocumentUri },
+                    Kind = GeneratedDocumentKind.Html
+                };
+
+                var response = _joinableTaskFactory.Run(async () =>
+                {
+                    var lspResponse = await _lspRequestInvoker.ReinvokeRequestOnServerAsync<DocumentContentsRequest, DocumentContentsResponse>(
+                        _activeWpfTextView.TextBuffer,
+                        "razor/generatedDocumentContents",
+                        RazorLSPConstants.RoslynLanguageServerName,
+                        request,
+                        CancellationToken.None);
+
+                    return lspResponse?.Response;
+                });
+
+                if (response != null)
+                {
+                    OpenGeneratedCode(response.FilePath, response.Contents);
+                    return;
+                }
+            }
+        }
+
+        // Fall back to legacy method if cohosting is not enabled or failed
```

Comment:
This code can actually be removed, is should be unnecessary. I was wrong before. The existing OpenVirtualDocuments call should work just fine in cohosting for Html.